src/main/java/dwbh/api/util/Result.java

```diff
@@ -64,6 +66,96 @@ public T getSuccess() {
     return errorList;
   }
 
+  /**
+   * Whether the instance is a success
+   *
+   * @return true if the result doesn't have any error
+   * @since 0.1.0
+   */
+  public boolean isSuccess() {
+    return !this.hasErrors();
+  }
+
+  /**
+   * Whether the instance has errors or not
+   *
+   * @return true if the result does have errors
+   * @since 0.1.0
+   */
+  public boolean hasErrors() {
+    return Optional.ofNullable(this.errorList).map(list -> !list.isEmpty()).orElse(false);
+  }
+
+  /**
+   * This method creates a possible failing {@link Result} from a given {@link Check}. Once you have
+   * the result you can chain calls to {@link Result#thenCheck(Supplier)} if you want to check more
+   * conditions or simply call {@link Result#then(Supplier)} if there're no more checkers and you
+   * want to end the execution with the final success result.
+   *
+   * @param <U> the expected result type
+   * @param check check to start a {@link Result} chain
+   * @return a partial and possibly failing {@link Result}
+   * @since 0.1.0
+   */
+  public static <U> Result<U> fromCheck(Check check) {
```

Comment:
I think that pass the first Check here, and that is is a Check and not a Supplier,  is a bit inconsistent.

I suggest to replace this method for:

```
public static <U> Result<U> build() {
    return Result.result(null);
}
```
And add the first check with `thenCheck`, like the rest.

Reply:
I agree. But instead of `build()` I would use `create()` because in builders we call `build()` at the end of the chain.

Reply:
Changed to `create()`